### ISSUE TITLE
Support for D versions + other flags with the DMD linter

### ISSUE THREAD

**Issue by rtbo** [dense-analysis/ale#3637]
Some code rely on some versions passed to DMD. For example:
```d
module bindbc.lua.v53.bindstatic;

version(BindBC_Static) version = BindLua_Static;
version(BindLua_Static) {
    version(LUA_53) version = LUA_53_STATIC;
}

version(LUA_53_STATIC):
// lua-5.3 static bindings follow
```
To use this code I define versions `LUA_53` and `BindBC_Static` in my `dub.json` source file. 
The ALE linter is not using it and it creates false lint errors when I use lua symbols in my code.

The versions can be found from `dub describe --data=versions`. Versions is not the only thing that can create false lint errors.
There are also debug versions, imported files, imported string files.

My suggestion would be to replace the current `dub describe --import-paths` with `dub describe --data=import-paths --data=string-import-paths --data=import-files --data=versions --data=debug-versions` and to parse the output of that command to retrieve all the options to give to DMD for linting.
